src/storage/compactor/src/server.rs

```diff
@@ -77,9 +77,11 @@ pub async fn compactor_serve(
     // limited at first.
     let storage_config = Arc::new(config.storage);
     let state_store_stats = Arc::new(StateStoreMetrics::new(registry.clone()));
+    let state_store_url = cluster_config.state_store_url.as_str();
+    assert_eq!(state_store_url, opts.state_store);
     let object_store = Arc::new(
         parse_remote_object_store(
-            opts.state_store
+            state_store_url
```

Comment:
@hzxa21  Since there are some users using our system, shall this configuration consider compatibility ?

Reply:
RIght, perhaps we should change `assert` to a warning log.

Reply:
If the change only affects how to start the processes, I think it is fine since it doesn't affect data/metadata. It is easy to guide our user to change the way they boot the processes as long as user doesn't need to clean or migrate their data.